x-pack/plugins/apm/public/selectors/latency_chart_selectors.ts

```diff
@@ -30,19 +31,27 @@ export function getLatencyChartSelector({
   theme: EuiTheme;
   latencyAggregationType?: string;
 }): LatencyChartData {
-  if (!latencyChart?.latencyTimeseries || !latencyAggregationType) {
+  if (
+    !latencyChart?.currentPeriod.latencyTimeseries ||
+    !latencyAggregationType
+  ) {
     return {
-      latencyTimeseries: [],
+      currentPeriod: undefined,
+      previousPeriod: undefined,
       mlJobId: undefined,
       anomalyTimeseries: undefined,
     };
   }
```

Comment:
Should be able to avoid the explicit `undefined` assignment for properties by using `Partial` in the return type:

```typescript
export interface LatencyChartData {
  currentPeriod: APMChartSpec<Coordinate>;
  previousPeriod: APMChartSpec<Coordinate>;
  mlJobId: string;
  anomalyTimeseries: { boundaries: APMChartSpec[]; scores: APMChartSpec };
}

export function getLatencyChartSelector({
  latencyChart,
  theme,
  latencyAggregationType,
}: {
  latencyChart?: LatencyChartsResponse;
  theme: EuiTheme;
  latencyAggregationType?: string;
}): Partial<LatencyChartData> {
  if (
    !latencyChart?.currentPeriod.latencyTimeseries ||
    !latencyAggregationType
  ) {
    return {};
  }
```